contracts/UniswapV3Staker.sol

```diff
@@ -131,14 +130,16 @@ contract UniswapV3Staker is
 
         Incentive memory incentive = incentives[key];
         require(incentive.rewardToken != address(0), 'INVALID_INCENTIVE');
-        delete incentives[key];
 
         // TODO: check for possible failures
-        IERC20Minimal(params.rewardToken).transfer(
+        TransferHelper.safeTransfer(
+            params.rewardToken,
             msg.sender,
             incentive.totalRewardUnclaimed
         );
 
+        delete incentives[key];
```

Comment:
Moved the delete to be after the `safeTransfer`. The assumption is that if the transfer fails, incentives will still have the key intact so that the transfer can be re-tried.

Reply:
the whole transaction will be reverted if the transfer fails, including the key deletion, right?